cfpb/cfgov-refresh
# Hide unpublished jobs in jobs list/table

The new Wagtail jobs list/table blocks introduced in #2341 show all jobs that are currently open, but fail to hide any `JobListingPage` pages that aren't yet published to the live site (those that are in draft or shared mode). This PR introduces logic to filter these out.

## Changes

- Additional filter that enforces `live=True` in the `OpenJobListingsMixin` that the table/list blocks use to generate their `QuerySet` of `JobListingPage` objects.

## Testing

- Run `tox -e fast` and see new tests in `jobmanager.tests.models.test_blocks`.
- To test the `JobListingTable`:
 1. Run a local server and create a new `BrowsePage`, then add a `JobListingTable`.
 1. Create a new `JobListingPage` under the `CFGOV` root, and select "Save as Draft" (make sure your job open/close dates make it currently open).
 1. Note when you refresh your `BrowsePage` that the job does not show up in the table. 
 1. Now go back to your job page and select "Share on Content".
 1. Note that the job still does not appear when the table is refreshed.
 1. Now go back to your job page and select "Publish to WWW".
 1. You should see your new job appear in the table.
- To test the `JobListingList`:
 1. Repeat the above steps but with a `SublandingPage` with a `JobListingList` (in the sidebar).

## Review

- @richaagarwal @cfpb/cfgov-backends 

## Checklist
